packages/grafana-runtime/src/services/dashboard-srv/types.ts

```diff
@@ -0,0 +1,123 @@
+import { PanelModel as BaseCorePanelModel } from '@grafana/data';
+import { GridPos } from '@grafana/schema';
+
+// Middle types to glue Public Plugin APIs types with internal Grafana types
+export interface CorePanelModel extends BaseCorePanelModel {
+  gridPos: GridPos;
+  refresh: () => void;
+}
+
+export interface CoreDashboardModel {
+  panels: CorePanelModel[];
+  uid: string;
+  title: string;
+  updatePanels: (panels: CorePanelModel[]) => unknown;
+}
+
+export interface CoreDashboardSrv {
+  dashboard?: CoreDashboardModel;
+}
+
+// Public types
+
+/**
+ * Used to interact with the current Grafana dashboard.
+ *
+ * @public
+ */
+export interface PluginsAPIDashboardSrv {
+  /**
+   * @deprecated use `getCurrentDashboard()` instead
+   */
+  dashboard?: PluginsAPIDashboardModel;
```

Comment:
I don't think we need this, getCurrentDashboard should be enough

Reply:
I agree. This is to help the twinmaker app and other plugins keep backwards compatibility without having to modify their original code but more and more I think it is a bad idea.

Reply:
but the interface does not look backward compatible (getCurrent vs getCurrentDashboard)

Reply:
@torkelo that's a good point. Most likely the plugins will have to deal with some types problems that can be resolved with some types assertions. A small price to pay for not having to rewrite their whole code.

I think it is best to just remove these deprecated methods. I will just do that, too much trouble to try to create something new and keep the old working too.